kit/ChildSession.cpp

```diff
@@ -543,9 +543,17 @@ bool ChildSession::_handleInput(const char *buffer, int length)
             }
             else if (tokens[1].find(".uno:Save") != std::string::npos)
             {
-                // Disable processing of other messages while saving document
-                InputProcessingManager processInput(getProtocol(), false);
-                return unoCommand(tokens);
+                static bool doBgSave = !!getenv("COOL_BGSAVE");
+
+                // Attempt to save in the background.
+                if (doBgSave && !saveDocumentAsync(tokens))
+                { // fallback to synchronous save
+
+                    // Disable processing of other messages while saving document
```

Comment:
I thought ".uno:Save" returned after queuing it, and it was handled in the background. Perhaps I misunderstand how saving is handled in Core.

Reply:
Hmm; synchronous save is perhaps misleading though it mostly happens in that callback - the final status thing comes later (I think).